src/app/core/params.ts

```diff
@@ -4,8 +4,8 @@ import Parameter from "./parameter";
  * Params class
  */
 export default class Params {
-    private unique: Array<string>;
-    private heritable: Array<string>;
+    private unique: {[index: string]: string};
+    private heritable: {[index: string]: string};
```

Comment:
@MurrayLF Why are we moving away from arrays here ? I belive this was our decision to change it

Reply:
They have been rewritten to be Maps. Arrays are indexed into only with numbers, to index with strings you need a map